contracts/contracts/oracle/BaseOracle.sol

```diff
@@ -0,0 +1,202 @@
+    }
+
+    /***************************************
+                Internal Setters
+    ****************************************/
+
+    /// @notice Sets the contract or account that can add Oracle prices
+    /// @param _oracleUpdater Address of the contract or account that can update the Oracle prices
+    function _setOracleUpdater(address _oracleUpdater) internal {
+        emit SetOracleUpdater(_oracleUpdater);
+        oracleUpdater = _oracleUpdater;
+    }
+
+    /***************************************
+                External Setters
+    ****************************************/
+
+    /// @notice Sets the contract or account that can update the Oracle prices
+    /// @param _oracleUpdater Address of the contract or account that can update the Oracle prices
+    function setOracleUpdater(address _oracleUpdater) external onlyGovernor {
+        _setOracleUpdater(_oracleUpdater);
+    }
+
+    /***************************************
+                Metadata
+    ****************************************/
+
+    /// @notice The number of decimals in the Oracle price.
+    function decimals()
+        external
+        pure
+        virtual
+        override
+        returns (uint8 _decimals)
+    {
+        _decimals = 18;
+    }
+
+    /// @notice The version number for the AggregatorV3Interface.
+    /// @dev Adheres to AggregatorV3Interface, which is different than typical semver
+    function version()
+        external
+        view
+        virtual
+        override
+        returns (uint256 _version)
+    {
+        _version = 1;
+    }
+
+    /***************************************
+                Oracle Receiver
+    ****************************************/
+
+    /// @notice Adds a new Oracle price by the Oracle updater.
+    /// Can not be run twice in the same block.
+    /// @param _price is the Oracle price with 18 decimals
+    function addPrice(uint128 _price) external override {
+        if (msg.sender != oracleUpdater) revert OnlyOracleUpdater();
+        if (_price == 0) revert NoPriceData();
+
+        // Can not add price in the same or previous blocks
+        uint256 _roundsLength = rounds.length;
+        if (
+            _roundsLength > 0 &&
+            block.timestamp <= rounds[_roundsLength - 1].timestamp
+        ) {
+            revert AddPriceSameBlock();
+        }
+
+        lastCorrectRoundId = uint80(_roundsLength);
+
+        rounds.push(
+            Round({ price: _price, timestamp: uint40(block.timestamp) })
+        );
+    }
+
+    /***************************************
+                Prices
+    ****************************************/
+
+    function _getRoundData(uint80 _roundId)
+        internal
+        view
+        returns (
+            uint80 roundId,
+            int256 answer,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        )
+    {
+        if (rounds.length <= _roundId) revert NoPriceData();
```

Comment:
Is this check built into new solidity versions on storage array access? 

> Accessing an array past its end causes a failing assertion.

Reply:
Solidity will revert. This code is to throw the more usable `NoPriceData` error.